src/colvar_UIestimator.h

```diff
@@ -0,0 +1,831 @@
+        UIestimator(const std::vector<double> lowerboundary,
+            const std::vector<double> upperboundary,
+            const std::vector<double> width,
+            const std::vector<double> krestr,                // force constant in eABF
+            const std::string& output_filename,              // the prefix of output files
+            const int output_freq,
+            const bool restart,                              // whether restart from a .count and a .grad file
+            const std::vector<std::string> input_filename,   // the prefixes of input files
+            const double temperature)
+        {
+
+            // initialize variables
+            this->lowerboundary = lowerboundary;
+            this->upperboundary = upperboundary;
+            this->width = width;
+            this->krestr = krestr;
+            this->output_filename = output_filename;
+            this->output_freq = output_freq;
+            this->restart = restart;
+            this->input_filename = input_filename;
+            this->temperature = temperature;
+
+            dimension = lowerboundary.size();
+
+            for (int i = 0; i < dimension; i++)
+            {
+                sum_x.push_back(n_vector<double>(lowerboundary, upperboundary, width, Y_SIZE, 0.0));
+                sum_x_square.push_back(n_vector<double>(lowerboundary, upperboundary, width, Y_SIZE, 0.0));
+
+                x_av.push_back(n_vector<double>(lowerboundary, upperboundary, width, Y_SIZE, 0.0));
+                sigma_square.push_back(n_vector<double>(lowerboundary, upperboundary, width, Y_SIZE, 0.0));
+            }
+
+            count_y = n_vector<int>(lowerboundary, upperboundary, width, Y_SIZE, 0);
+            distribution_x_y = n_matrix(lowerboundary, upperboundary, width, Y_SIZE);
+
+            grad = n_vector<std::vector<double> >(lowerboundary, upperboundary, width, 1, std::vector<double>(dimension, 0.0));
+            count = n_vector<int>(lowerboundary, upperboundary, width, 1, 0);
+
+            written = false;
+            written_1D = false;
+
+            if (dimension == 1)
+            {
+                std::vector<double> upperboundary_temp = upperboundary;
+                upperboundary_temp[0] = upperboundary[0] + width[0];
+                oneD_pmf = n_vector<double>(lowerboundary, upperboundary_temp, width, 1, 0.0);
+            }
+
+            if (restart == true)
+            {
+                input_grad = n_vector<std::vector<double> >(lowerboundary, upperboundary, width, 1, std::vector<double>(dimension, 0.0));
+                input_count = n_vector<int>(lowerboundary, upperboundary, width, 1, 0);
+
+                // initialize input_Grad and input_count
+                std::vector<double> loop_flag(dimension, 0);
+                for (int i = 0; i < dimension; i++)
+                {
+                    loop_flag[i] = lowerboundary[i];
+                }
+                while (true)
```

Comment:
These loops look like they could be written without gotos, just replacing both `while(true) `with `while(i >= 0)` -- or am I missing something?

Same remarks for the similar loops below.